src/dataframe/dataframe.jl

```diff
@@ -216,6 +215,21 @@ struct DataFrame <: AbstractDataFrame
     end
 end
 
+function _preprocess_column(col, len, copycols)
+    # check for vectors first as they are most common
+    col isa AbstractRange && return collect(col)
+    col isa AbstractVector && return copycols ? copy(col) : col
```

Comment:
When `copycols=false`, this function does very little work, so it would make sense to disable multithreading in that case.

Reply:
done